postgres.go

```diff
@@ -49,3 +49,34 @@ func StoreDB(c *gin.Context, pg_conn *gorm.DB, inf interface{}) bool {
 	}
 	return true
 }
+
+// Finds Data Entry with condition and updates it with new data
+func UpdateDB(c *gin.Context, pg_conn *gorm.DB, inf interface{}, condition interface{}) bool {
+	if err := pg_conn.Where(condition).Updates(inf).Error; err != nil {
+		c.AbortWithStatus(http.StatusInternalServerError)
+		LogError("FoxKit", err)
+		return false
+	}
+	return true
+}
+
+// Finds and Returns true with the Data Entry or false if not found
+func FindDB(c *gin.Context, pg_conn *gorm.DB, inf interface{}, condition interface{}) (bool, interface{}) {
+	data := pg_conn.Where(condition).First(inf)
```

Comment:
`func (*gorm.DB).First(dest interface{}, conds ...interface{}) (tx *gorm.DB)` takes `dest interface{}` which means that the data gorm is retrieving from the DB is stored into the struct that is passed as parameter. `data` in this case is just information about the DB request, so we just need to check if an error happened by:
```go
if err := pg_conn.Where(condition).First(inf).Error; err != nil {
  return err
}
```

That the function is able to pass the data directly into the struct given as parameter is possible through a simple pointer. That means that you actually pass the memory address to the struct to our function (`FindDB(..., &Struct)`) and then gorm can write into there.